src/app/services/command.service.ts

```diff
@@ -0,0 +1,115 @@
+import { Injectable } from '@angular/core';
+
+export interface Command {
+  cmd: string;
+  /* fullCMD is the full text, parameters are the parameters, parsed by using space as delimiter.
+     parameters[0] is therefore the command name */
+  function: (fullCMD: string, parameters: string[]) => void;
```

Comment:
Please change to something like 'rawString', 'rawCMD' or something like that as it seemed confusing to me.

Reply:
also outdated